doc/book/sql-ddl.md

```diff
@@ -87,8 +87,21 @@ $table->changeColumn('name', Column\Varchar('new_name', 50));
 You may also *drop* existing columns or constraints:
 
 ```php
+use \Zend\Db\Metadata\Object\ConstraintObject;
+
 $table->dropColumn('foo');
-$table->dropConstraint('my_index');
+
+$constraint = new ConstraintObject('my_index', null);
+$table->dropConstraint($constraint);
+```
+
+Notice: On MySQL, you need to specify the type of constraint you want to drop:
+```php
+use \Zend\Db\Metadata\Object\ConstraintObject;
+
+$fkConstraint = new ConstraintObject('my_fk', null);
+$fkConstraint->setType('FOREIGN KEY');
```

Comment:
> Or clean creation of ConstraintObject to create ForeignKey, which would be further reaching change.

`\Zend\Db\Metadata\Source\AbstractSource::getConstraint()` seems to be doing its job when analyzing the table and assigns the type of constraints correctly.

But yeah, when you dig that down you realize there are some nasty inconsistencies to take care of ahead of that bug.

Anyway, I'll take your remarks into account and I'll make some changes, hopefully for the best. But I'm not going to rewrite half of the framework either.